### PR TITLE
A lighter-touch fix for console focus.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Fix for #4837, alternative to #4886.

@jasongrout, can you comment on whether this interferes with #4330?

TLDR: there was a race condition in the console focusing for a new prompt cell.

The problem comes down to the difference in processing of a forced console execution vs an unforced one. The forced one is processed synchronously, the unforced one is processed asynchronously. The difference in control flow is as such:

#### Forced
1. A console command is executed, [getCurrent](https://github.com/jupyterlab/jupyterlab/blob/master/packages/console-extension/src/index.ts#L304-L311) is called. *This function sends and activate request to the console, which is processed asynchronously*.
2. The code is sent to the kernel, and a new prompt cell is created.
3. The `activate` request is handled later, and the new prompt cell is focused.

#### Unforced
1. 1. A console command is executed, [getCurrent](https://github.com/jupyterlab/jupyterlab/blob/master/packages/console-extension/src/index.ts#L304-L311) is called. 
2. The console checks with the kernel whether it should run code.
3. The `activate` request is handled, it seems in practice before the above check is complete. This activates the old prompt.
4. Later, the console learns it should execute, sends code to the kernel, and a new prompt cell is created.
5. The new prompt cell never gets an activate request.

As an aside, I have been bit by this auto-focusing side-effect of `getCurrent` before. Perhaps we should rethink that behavior?

### CODE REVIEW

**Comment by jasongrout** [comment]
I deleted the focusing behavior of getCurrent (in the notebook?) for #4330, but then put it back in because I thought it was too fundamental of a change in a tricky part of the codebase (handling focus) to do right before an RC.

Thanks for tracking this down, your explanation makes sense. I won't be able to test the PR until next week, but looking at the changes, they look good to me!

You can test #4330 by adding a keybinding for the new command there, and running that keybinding in the notebook, which should open a console *without* activating it. This keybinding should work, for example:

```
{
  "notebook:run-selected-or-current-line-in-console": {
    "command": "notebook:run-in-console",
    "keys": ["Ctrl G"],
    "selector":  ".jp-Notebook.jp-mod-editMode"
  }
}
```

**Comment by ian-r-rose** [comment]
Thanks @jasongrout, I just tested #4330 and it appears to have the right focusing behavior.

**Comment by blink1073** [comment]
Thanks!
